nengo_loihi/hardware/interface.py

```diff
@@ -582,25 +599,33 @@ def create_io_snip(self):
         self.host_socket_port = host_socket_port
 
         # --- create channels
-        size = (
+        input_channel_size = (
             1  # first int stores number of spikes
             + self.snip_max_spikes_per_step * SpikePacker.size()
             + total_error_len
         )
-        logger.debug("Creating nengo_io_h2c channel (%d)" % size)
+        logger.debug("Creating nengo_io_h2c channel (%d)" % input_channel_size)
         self.nengo_io_h2c = d_get(self.nxsdk_board, b"Y3JlYXRlQ2hhbm5lbA==")(
-            b"nengo_io_h2c",  # name of the channel
+            b"nengo_io_h2c",  # channel name
             **{
-                d(b"bnVtRWxlbWVudHM="): size,  # size of the channel (in elements)
-                d(b"bWVzc2FnZVNpemU="): 4,  # size of one packet (in bytes)
+                # channel size (in elements)
+                d(b"bnVtRWxlbWVudHM="): input_channel_size,
+                # size of one packet (in bytes)
+                d(b"bWVzc2FnZVNpemU="): 4 * self.channel_packet_size,
+                # size of send/receive buffer on chip/host (in packets)
+                d(b"c2xhY2s="): 16,
             },
         )
         logger.debug("Creating nengo_io_c2h channel (%d)" % n_outputs)
         self.nengo_io_c2h = d_get(self.nxsdk_board, b"Y3JlYXRlQ2hhbm5lbA==")(
-            b"nengo_io_c2h",  # name of the channel
+            b"nengo_io_c2h",  # channel name
             **{
-                d(b"bnVtRWxlbWVudHM="): n_outputs,  # size of the channel (in elements)
-                d(b"bWVzc2FnZVNpemU="): 4,  # size of one packet (in bytes)
+                # channel size (in elements)
+                d(b"bnVtRWxlbWVudHM="): n_outputs,
+                # size of one packet (in bytes)
+                d(b"bWVzc2FnZVNpemU="): 4 * self.channel_packet_size,
```

Comment:
The multiplication of size with 4 can be replaced with multiplication with a constant describing 4 e.g. `bytes_per_int32` or something better. (line 614, 626)

Reply:
`channel_bytes_per_element`